src/protocols/SamplesPacket.h

```diff
@@ -104,14 +177,13 @@ template<uint8_t chCount> class SamplesPacket
     uint8_t* tail[chCount];
     uint8_t* channel[chCount];
 
+    uint32_t offset;
+    uint32_t length;
+    uint32_t mCapacity;
+    uint8_t frameSize;
+
   public:
-    int64_t timestamp;
-    uint16_t offset;
-    uint16_t length;
-    uint16_t mCapacity;
-    uint16_t frameSize;
-    bool useTimestamp;
-    bool flush;
+    SDRDevice::StreamMeta metadata; ///< The stream metadata of the packet.
```

Comment:
SamplesPacket itself is a standalone data structure, it does not need to know about SDRDevice, or be coupled to it's meta data (which could contain data not relevant to the packet).